setup.py

```diff
@@ -97,45 +91,82 @@ def glob_locale_files(path: Path):
     return ret
 
 
-setup(
-    name="Red-DiscordBot",
-    version=get_version(),
-    packages=get_package_list(),
-    package_data=find_locale_folders(),
-    include_package_data=True,
-    url="https://github.com/Cog-Creators/Red-DiscordBot",
-    license="GPLv3",
-    author="Cog-Creators",
-    author_email="",
-    description="A highly customizable Discord bot",
-    classifiers=[
-        "Development Status :: 4 - Beta",
-        "Framework :: AsyncIO",
-        "Framework :: Pytest",
-        "Intended Audience :: Developers",
-        "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
-        "Operating System :: OS Independent",
-        "Programming Language :: Python :: 3.6",
-        "Topic :: Communications :: Chat",
-        "Topic :: Documentation :: Sphinx",
-    ],
-    entry_points={
-        "console_scripts": [
-            "redbot=redbot.__main__:main",
-            "redbot-setup=redbot.setup:main",
-            "redbot-launcher=redbot.launcher:main",
+if __name__ == "__main__":
+    if not check_compiler_available():
+        requirements.remove(
+            next(r for r in requirements if r.lower().startswith("python-levenshtein"))
+        )
+
+    if "READTHEDOCS" in os.environ:
+        requirements.remove(next(r for r in requirements if r.lower().startswith("discord.py")))
+
+    setup(
+        name="Red-DiscordBot",
+        version=get_version(),
+        packages=get_package_list(),
+        package_data=find_locale_folders(),
+        include_package_data=True,
+        url="https://github.com/Cog-Creators/Red-DiscordBot",
+        license="GPLv3",
+        author="Cog-Creators",
+        author_email="",
+        description="A highly customizable Discord bot",
+        classifiers=[
+            "Development Status :: 4 - Beta",
+            "Framework :: AsyncIO",
+            "Framework :: Pytest",
+            "Intended Audience :: Developers",
+            "License :: OSI Approved :: GNU General Public License v3 or later (GPLv3+)",
+            "Operating System :: OS Independent",
+            "Programming Language :: Python :: 3.6",
```

Comment:
Since this will allow 3.7 usage, this should also include a 3.7 identifier for pypi